clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -769,13 +770,13 @@ public void testRename_existingFileToExistingDirName() throws ApiException, IOEx
 
         boolean renamed = fs.rename(src, dst);
         Assert.assertTrue(renamed);
-        Path expectedDstPath = new Path("lakefs://repo/main/existing-dir2/existing-dir1/existing.src");
+        Path expectedDstPath = new Path("lakefs://repo/main/existing-dir2/existing.src");
         Assert.assertTrue(dstPathLinkedToSrcPhysicalAddress(srcObjLoc, fs.pathToObjectLocation(expectedDstPath)));
         verifyObjDeletion(srcObjLoc);
     }
 
     /**
-     * rename(srcDir(containing srcDir/a.txt, srcDir/b.txt), non-existing-dstdir) -> non-existing-dstdir/a.txt, non-existing-dstdir/b.txt
+     * rename(srcDir(containing srcDir/a.txt, srcDir/b.txt), non-existing-dir/new) -> unsupported, rename should fail by returning false
      */
     @Test
     public void testRename_existingDirToNonExistingDirName() throws ApiException, IOException {
```

Comment:
As discussed f2f, you can add another test case that tests rename to a non-existing dir with an existing parent (parent can be the branch)

Reply:
Adding another test